include/crm/common/ordering.h

```diff
@@ -37,6 +37,12 @@ enum pe_ordering {
      */
     pcmk_ord_then_implies_first         = (1U << 4),
 
+    /*!
+     * If 'first' is required, 'then' becomes required; if 'first' is a stop of
+     * a blocked resource, 'then' becomes unrunnable
```

Comment:
If both are required, 'first' is a necessary condition for 'then'

Reply:
That would make sense for `pcmk_ord_enabled|pcmk_ord_first_implies_then`. If 'first' is required, 'then' is required and must occur *after* 'first' because ordering is enabled. If 'first' is blocked, 'then' is unrunnable because 'first' can't complete.

It makes somewhat less sense to me for `pcmk_ord_first_implies_then` by itself. That flag reads like it should indicate that 'first' required implies 'then' required and nothing else.

---

There's a stronger case to be made for "'first' blocked -> 'then' unrunnable" with the `pcmk_ord_then_implies_first` flag (compared to the `pcmk_ord_first_implies_then` flag. With "then implies first", we have:
```
if 'then', then 'first'
not 'first'
therefore, not 'then'
```

The equivalent for "first implies then" would be to make 'first' unrunnable if 'then' is blocked, which we don't do.
```
if 'first', then 'then'
not 'then'
therefore, not 'first'
```

---

None of this is particularly clear-cut. Just my interpretations. Largely moot anyway due to backward compatibility.